assets/js/favorites.js

```diff
@@ -0,0 +1,40 @@
+var favoritesList;
+
+function init() {
+    favoritesList = localStorage.getItem("favorites");
+    if (favoritesList) {
+        favoritesList = JSON.parse(favoritesList);
+    } else {
+        favoritesList = [];
+    }
+}
+
+function displayFavorites() {
+    var favoritesContainer = document.getElementById("favorites-list");
+    favoritesContainer.innerHTML = "";
+
+    if (favoritesList.length > 0) {
+        favoritesList.forEach(function (songIndex) {
+            var song = lastSearch[songIndex].track;
```

Comment:
lastSearch was defined in index.html but not favorites.html, so you might need to add localStorage for this item